### PR TITLE
feat: Update contract definitions #122

### PR DESCRIPTION
**Pull request by David-Sod**

### CODE REVIEW

File: src/contracts/definitions/snip-721-def.ts [hunk 1]
```diff
@@ -12,17 +12,22 @@ export const snip721Def: ContractDefinition = {
       return { contract_info: {} };
     },
 
-    getNumTokens({ address, key: viewing_key }: Context): ContractQueryRequest {
-      const viewer = { address, viewing_key };
+    getNumTokens(
+      { address, key: viewing_key }: Context,
+      sendViewer?: boolean
+    ): ContractQueryRequest {
+      let viewer = sendViewer ? { address, viewing_key } : null;
```
Comment (luis-stakeordie) [inline src/contracts/definitions/snip-721-def.ts:19]: `let` is used for mutable variables... why don't use `const` for immutable?

File: src/contracts/definitions/snip-721-def.ts [hunk 2]
```diff
@@ -82,12 +91,14 @@ export const snip721Def: ContractDefinition = {
     },
 
     getApprovedForAll(
-      { address: owner }: Context,
+      { address: owner, key }: Context,
       include_expired?: boolean
     ): ContractQueryRequest {
+      let viewing_key = key ? key : null;
```
Comment (luis-stakeordie) [inline src/contracts/definitions/snip-721-def.ts:97]: Do we need to explicitly pass `null`? Can't we just simply rename `key` to `viewing_key`?

**Review by luis-stakeordie — CHANGES_REQUESTED** [review]
Is the `snip20Def` ok?

Check out other comments too, thanks!

File: src/contracts/definitions/snip-721-def.ts (hunk 2 above)
Comment (David-Sod) [reply src/contracts/definitions/snip-721-def.ts:97]: yeah, we need to be explicit, if we pass `undefined` or `""` the contract thinks is a vk and tries to validate it (obviously that fails)

File: src/contracts/definitions/snip-721-def.ts (hunk 1 above)
Comment (David-Sod) [reply src/contracts/definitions/snip-721-def.ts:19]: no reason, we can switch to `const`, working on it

Comment (David-Sod) [reply src/contracts/definitions/snip-721-def.ts:19]: Yeah, all queries and messages are complete and there is no optional VK so we need to pass always a VK

File: src/contracts/definitions/snip-721-def.ts (hunk 2 above)
Comment (luis-stakeordie) [reply src/contracts/definitions/snip-721-def.ts]: Understood!

**Review by luis-stakeordie — APPROVED** [review]
Looks good, thanks @David-Sod!

**Pull request merged**